ct.py

```diff
@@ -12,19 +12,22 @@
 from CTBASIC.graphics import CLS_TEK as CLS
 
 CT = '01;'
-OUTPUT = re.compile(r'1000000100(1[01]{8}0)*1000000110$')
+STX = '1000000100'
+ETX = '1000000110'
+OUTPUT = re.compile(STX + r'(1[01]{8}0)*' + ETX + r'$')
 
 
 def output(s):
-    if not s.endswith('1000000110'):
-         return
-    m = OUTPUT.search(s)
+    if not s.endswith(ETX):
+        return
+    last = s.rfind(STX)
```

Comment:
this isn't working because it can incorrectly match non-aligned bits / frames.
Back to the drawing board.
I want to regex match multiples of 10 bits from the right end of the `ETX` terminated string  back to the first occurrence of `STX`.

Alternatively, I want to exclude any `STX` from the middle portion of the current regex:

Currently I can't think of a direct way to do this other than reversing both the string and the regex...
    OUTPUT = re.compile(STX + r'(1[01]{8}0)*' + ETX + r'$')

Reply:
latest commits resolve this by reversing the string and matching on a reversed version of the regex.

Apparently .net regexes and the Python regex module can perform RTL matches, but it is not a standard thing.